src/v/datalake/coordinator/catalog_factory.cc

```diff
@@ -7,15 +7,70 @@
  *
  * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
  */
+#include "datalake/coordinator/catalog_factory.h"
+
 #include "config/configuration.h"
 #include "datalake/logger.h"
 #include "iceberg/catalog.h"
 #include "iceberg/filesystem_catalog.h"
 #include "iceberg/rest_catalog.h"
 #include "iceberg/rest_client/catalog_client.h"
-#include "thirdparty/ada/ada.h"
+#include "net/tls.h"
+#include "net/tls_certificate_probe.h"
 
 #include <absl/strings/numbers.h>
+namespace {
+
+ss::future<ss::shared_ptr<ss::tls::certificate_credentials>>
+build_tls_credentials(
```

Comment:
For this we will require a whole bunch of new configuration options i.e.
```
iceberg_rest_catalog_trust_file: 
iceberg_rest_catalog_crl_file:
```
If we want to support `mTLS` we will need a key and cert files as well